core/test_blockchain.go

```diff
@@ -1187,6 +1193,16 @@ func TestReprocessAcceptBlockIdenticalStateRoot(t *testing.T, create func(db eth
 	if err != nil {
 		t.Fatal(err)
 	}
+	blockchainToStop := blockchain
+	alreadyStoped := false
+	blockchainStopOnlyOnce := func() {
+		if alreadyStoped {
+			return
+		}
+		alreadyStoped = true
+		blockchainToStop.Stop()
+	}
+	defer blockchainStopOnlyOnce()
```

Comment:
Could we change this to use `sync.Once` so that it's using an easy to understand primitive? Maybe something like this?

```suggestion
	var stopOnce sync.Once
	defer stopOnce.Do(blockchain.Stop())
	.
	.
	.
	stopOnce.Do(blockchain.Stop())
```

Reply:
done